contracts/peripheral/YieldProxy.sol

```diff
@@ -158,394 +52,71 @@ contract YieldProxy is DecimalMath {
-        IFYDai fyDai = pool.fyDai();
-        require(fyDai.isMature() != true, "YieldProxy: Only before maturity");
-        require(dai.transferFrom(msg.sender, address(this), daiUsed), "YieldProxy: Transfer Failed");
-
-        // calculate needed fyDai
-        uint256 daiReserves = dai.balanceOf(address(pool));
-        uint256 fyDaiReserves = fyDai.balanceOf(address(pool));
-        uint256 daiToAdd = daiUsed.mul(daiReserves).div(fyDaiReserves.add(daiReserves));
-        uint256 daiToConvert = daiUsed.sub(daiToAdd);
-        require(
-            daiToConvert <= maxFYDai,
-            "YieldProxy: maxFYDai exceeded"
-        ); // 1 Dai == 1 fyDai
-
-        // convert dai to chai and borrow needed fyDai
-        chai.join(address(this), daiToConvert);
-        // look at the balance of chai in dai to avoid rounding issues
-        uint256 toBorrow = chai.dai(address(this));
-        controller.post(CHAI, address(this), msg.sender, chai.balanceOf(address(this)));
-        controller.borrow(CHAI, fyDai.maturity(), msg.sender, address(this), toBorrow);
-        
-        // mint liquidity tokens
-        return pool.mint(address(this), msg.sender, daiToAdd);
-    }
-
-    /// @dev Burns tokens and sells Dai proceedings for fyDai. Pays as much debt as possible, then sells back any remaining fyDai for Dai. Then returns all Dai, and if there is no debt in the Controller, all posted Chai.
-    /// Caller must have approved the proxy using`controller.addDelegate(yieldProxy)` and `pool.addDelegate(yieldProxy)`
-    /// Caller must have approved the liquidity burn with `pool.approve(poolTokens)`
-    /// @param poolTokens amount of pool tokens to burn. 
-    /// @param minimumDaiPrice minimum fyDai/Dai price to be accepted when internally selling Dai.
-    /// @param minimumFYDaiPrice minimum Dai/fyDai price to be accepted when internally selling fyDai.
-    function removeLiquidityEarlyDaiPool(IPool pool, uint256 poolTokens, uint256 minimumDaiPrice, uint256 minimumFYDaiPrice) external {
-        onlyKnownPool(pool);
-        IFYDai fyDai = pool.fyDai();
-        uint256 maturity = fyDai.maturity();
-        (uint256 daiObtained, uint256 fyDaiObtained) = pool.burn(msg.sender, address(this), poolTokens);
-
-        // Exchange Dai for fyDai to pay as much debt as possible
-        uint256 fyDaiBought = pool.sellDai(address(this), address(this), daiObtained.toUint128());
-        require(
-            fyDaiBought >= muld(daiObtained, minimumDaiPrice),
-            "YieldProxy: minimumDaiPrice not reached"
-        );
-        fyDaiObtained = fyDaiObtained.add(fyDaiBought);
-        
-        uint256 fyDaiUsed;
-        if (fyDaiObtained > 0 && controller.debtFYDai(CHAI, maturity, msg.sender) > 0) {
-            fyDaiUsed = controller.repayFYDai(CHAI, maturity, address(this), msg.sender, fyDaiObtained);
-        }
-        uint256 fyDaiRemaining = fyDaiObtained.sub(fyDaiUsed);
+// proxy contract with no storage
+contract YieldProxy is YieldAuth {
```

Comment:
This contract seems to have the upgradable functions, but doesn't inherit from ´ProxyStorage.sol`. I'm confused.

Reply:
Explained below